crates/ironrdp-session/src/active_stage.rs

```diff
@@ -281,25 +284,25 @@ impl TryFrom<x224::ProcessorOutput> for ActiveStageOutput {
 
 /// Reasons for graceful disconnect. This type provides GUI-friendly descriptions for
 /// disconnect reasons.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub enum GracefulDisconnectReason {
     UserInitiated,
     ServerInitiated,
-    Other(&'static str),
+    Other(String),
 }
 
 impl GracefulDisconnectReason {
-    pub fn description(&self) -> &'static str {
+    pub fn description(&self) -> String {
         match self {
-            GracefulDisconnectReason::UserInitiated => "user initiated disconnect",
-            GracefulDisconnectReason::ServerInitiated => "server initiated disconnect",
-            GracefulDisconnectReason::Other(description) => description,
+            GracefulDisconnectReason::UserInitiated => "user initiated disconnect".to_owned(),
+            GracefulDisconnectReason::ServerInitiated => "server initiated disconnect".to_owned(),
+            GracefulDisconnectReason::Other(description) => description.clone(),
```

Comment:
Technically, many of these are _user-initiated disconnects_, but using `Other` is the easiest way to get a custom description..

Reply:
What are the other user-initiated disconnects?

Reply:
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/a21a1bd9-2303-49c1-90ec-3932435c248c

A lot of these might be considered user-initiated:
- ERRINFO_DISCONNECTED_BY_OTHERCONNECTION
- ERRINFO_RPC_INITIATED_DISCONNECT_BYUSER
- ERRINFO_LOGOFF_BY_USER
- ERRINFO_SERVER_SHUTDOWN
- ERRINFO_SERVER_REBOOT

That said, I think we'd be debating semantics, the goal is to give the user as specific a disconnect reason as possible, and I think we're accomplishing that as-is.